tests/timeline.py

```diff
@@ -0,0 +1,148 @@
+from datetime import datetime, timedelta, date
+
+from ics import Calendar, Event
+from ics.timezone import UTC
+
+
+def make_calendar():
+    """Create a test calendar for use in tests."""
+    cal = Calendar()
+    cal.events.extend([
+        Event("second", datetime(2000, 2, 1, 12, 0)),
+        Event("fourth", datetime(2000, 4, 1, 12, 0)),
+        Event("third", datetime(2000, 3, 1, 12, 0)),
+        Event("first", datetime(2000, 1, 1, 12, 0)),
+    ])
+    return cal
+
+
+def test_iteration():
+    """Test chronological iteration of a timeline.""" 
+    cal = make_calendar()
+    assert [e.summary for e in cal.timeline] == [
+        "first", "second", "third", "fourth"
+    ]
+
+def test_on():
+    """Test returning events on a particualr day."""
+    cal = make_calendar()
+    timeline = cal.timeline
+    assert [e.summary for e in cal.timeline.on(date(2000, 1, 1))] == ["first"]
+    assert [e.summary for e in cal.timeline.on(date(2000, 2, 1))] == ["second"]
+    assert [e.summary for e in cal.timeline.on(datetime(2000, 3, 1, 6, 0))] == [
+        "third"
+    ]
+
+
+def test_start_after():
+    """Test chronological iteration starting at a specific time."""
+    cal = make_calendar()
+    timeline = cal.timeline
+
+    assert [e.summary for e in cal.timeline.start_after(date(2000, 2, 1))] == [
+        "second", "third", "fourth"
+    ]
+
+
+def test_at():
+    """Test returning events at a specific time."""
+    cal = Calendar()
+    cal.events.extend([
+        Event(
+            "earlier",
+            begin=datetime(2000, 1, 1, 11, 0),
+            end=datetime(2000, 1, 1, 11, 30)
+        ),
+        Event("event",
+            begin=datetime(2000, 1, 1, 12, 0),
+            end=datetime(2000, 1, 1, 13, 0),
+        ),
+        Event("later",
+            begin=datetime(2000, 1, 2, 12, 0),
+            end=datetime(2000, 1, 2, 13, 0),
+        )
+    ])
+    assert [
+        e.summary for e in cal.timeline.at(datetime(2000, 1, 1, 11, 59))
+    ] == []
+    assert [
+        e.summary for e in cal.timeline.at(datetime(2000, 1, 1, 12, 0))
+    ] == ["event"]
+    assert [
+        e.summary for e in cal.timeline.at(datetime(2000, 1, 1, 12, 30))
+    ] == ["event"]
+    assert [
+        e.summary for e in cal.timeline.at(datetime(2000, 1, 1, 12, 59))
+    ] == ["event"]
+    assert [
+        e.summary for e in cal.timeline.at(datetime(2000, 1, 1, 13, 0))
+    ] == []
+
+
+def test_now():
```

Comment:
I'm not a huge fan of this test for 2 reasons:
* Using `datetime.now()` in tests usually breaks at some point, either because of DST, or something else
* You are not only testing `now()` but also `at()` in this test so if we break `at()` at some point, this test will fail too but `now()` would still be correct.

I see 2 options here:
1. Do not test this function, its a 1 line ( `return self.at(datetime.utcnow())`), what could go wrong ? 
2. Use something like [freezegun](https://github.com/spulec/freezegun) to fix the time, then mock  `at()` and verify the mock was indeed called with the frozen time you set.

If `now()` was a complicated function that we could mess up easily, i would go for option 2. But here, the test would be 10x longer than the actual function that is being tested and it would need a new dependency so i suggest to drop this test :)

Reply:
I went with #2 and i think combined with the move to use fixtures its a little less offensive.  That is, if we use fixtures for all the datetime based tests then this test doesn't really stick out as being overly complex use with freezegun. I also added a test for `today` while I was here.

If you have a look and still think its not worth having, then happy to drop, just let me know what you think.